x/airdrop/client/cli/parser.go

```diff
@@ -0,0 +1,95 @@
+package cli
+
+import (
+	"bufio"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	sdkmath "cosmossdk.io/math"
+
+	"github.com/Stride-Labs/stride/v22/x/airdrop/types"
+)
+
+// Parses an allocations CSV file consisting of allocations for various addresses
+//
+// Example Schema:
+//
+//	strideXXX,10,10,20
+//	strideYYY,0,10,0
+func ParseMultipleUserAllocations(fileName string) ([]types.RawAllocation, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	rows, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	allAllocations := []types.RawAllocation{}
+	for _, row := range rows {
+		if len(row) < 2 {
+			return nil, errors.New("invalid csv row")
+		}
+
+		userAddress := row[0]
+
+		allocations := []sdkmath.Int{}
+		for _, allocationString := range row[1:] {
+			allocation, ok := sdkmath.NewIntFromString(allocationString)
+			if !ok {
+				return nil, fmt.Errorf("unable to parse allocation %s into sdk.Int", allocationString)
+			}
+			allocations = append(allocations, allocation)
+		}
+
+		allocation := types.RawAllocation{
+			UserAddress: userAddress,
+			Allocations: allocations,
+		}
+		allAllocations = append(allAllocations, allocation)
```

Comment:
should we verify all the `allocation`s have the same length?

Reply:
it's handled in validate basic!

Reply:
Oh nice! Resolving